api/envoy/api/v2/endpoint/endpoint.proto

```diff
@@ -74,6 +74,11 @@ message LbEndpoint {
   // to subset the endpoints considered in cluster load balancing.
   core.Metadata metadata = 3;
 
+  // Used as the key to identiy the transport socket configuration defined
+  // in *Cluster.endpoint_transport_socket_map* to use when connecting to this endpoint.
+  // [#not-implemented-hide:]
+  string transport_socket_name = 6;
```

Comment:
fyi Istio uses the LbEndpoint for strict dns as well. so this should work there as well right?

Reply:
right.